compiler/one-cmds/onnx_legalizer.py

```diff
@@ -0,0 +1,749 @@
+#!/usr/bin/python3
+import onnx
+import onnx.numpy_helper
+import sys
+import numpy as np
+import re
+
+# Transform onnx model to make it compilable with our toolchain
+# List of transformations:
+# - Replace RNN operation with unrolled subgraph
+# - Replace LSTM operation with unrolled subgraph
+
+
+def reverse_str(s):
+    return ''.join(reversed(s))
+
+
+def parse_tensor_name(name):
+    rev = reverse_str(name)
+    m = re.match('(\d*)(.*)', rev)
+    if m.groups()[0] != '':
+        return (reverse_str(m.groups()[1]), int(reverse_str(m.groups()[0])))
+    else:
+        return (reverse_str(m.groups()[1]), 0)
```

Comment:
@binarman How about just do like this rather than reversing strings?
```python
m = re.findall('\D+|\d*', 'abc_123')
```

Is there any tricky case?

Reply:
Name in onnx is an arbitrary string, so "123" and "abc_321_cba_123" are valid.
This base name and id convention is my attempt to make new tensor names without risk of name collision.

Your approach do the trick, but requires some additional logic to cut only last number anyway.

If you think it is better option, I can replace reverse string approach with findall. reverse is just a first solution that came to my mind =)